### PR TITLE
Updates for Multi-NICs

### PR DESCRIPTION
**Pull request by Brian-McM**
Update pins and propogate MULTI_INTERFACE_MODE to the api server and query server

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/apiserver.go [hunk 1]
```diff
@@ -568,6 +568,16 @@ func (c *apiServerComponent) apiServerContainer() corev1.Container {
 
 	isPrivileged := true
 
+	env := []corev1.EnvVar{
+		{Name: "DATASTORE_TYPE", Value: "kubernetes"},
+	}
+
+	if c.installation.Spec.Variant == operator.TigeraSecureEnterprise {
```
Comment (caseydavenport) [inline pkg/render/apiserver.go:575]: The apiserver is an enterprise only component, so don't need this check.

Comment (caseydavenport) [reply pkg/render/apiserver.go:575]: Also, we should prevent setting `MultiInterfaceMode` to non-nil for Calico OSS clusters in validation anyway

Comment (Brian-McM) [reply pkg/render/apiserver.go:575]: Right, I think we do but I'll double check

Comment (Brian-McM) [reply pkg/render/apiserver.go]: We don't actually, but I'll make a separate PR to handle that.

**Pull request merged**